src/kv/mod.rs

```diff
@@ -1,35 +1,104 @@
+mod cache;
 /// Etcd delete mod for delete operations.
 mod delete;
+/// Etcd get mod for get operations.
+mod get;
 /// Etcd put mod for put operations.
 mod put;
 /// Etcd range mod for range fetching operations.
 mod range;
 /// Etcd txn mod for transaction operations.
 mod txn;
 
-use super::OverflowArithmetic;
+pub use super::watch::EtcdWatchRequest;
+pub use cache::Cache;
 pub use delete::{EtcdDeleteRequest, EtcdDeleteResponse};
+pub use get::{EtcdGetRequest, EtcdGetResponse};
 pub use put::{EtcdPutRequest, EtcdPutResponse};
 pub use range::{EtcdRangeRequest, EtcdRangeResponse};
 pub use txn::{EtcdTxnRequest, EtcdTxnResponse, TxnCmp, TxnOpResponse};
 
-use crate::protos::rpc_grpc::KvClient;
-
-use crate::protos::kv;
+use super::OverflowArithmetic;
+use crate::protos::kv::Event_EventType;
+use crate::protos::rpc::RangeResponse;
+use crate::protos::rpc_grpc::{KvClient, WatchClient};
+use futures::stream::StreamExt;
+use grpcio::WriteFlags;
+use log::warn;
+use protobuf::RepeatedField;
+use std::str;
+
+use crate::protos::kv::KeyValue;
 use crate::Result as Res;
+use futures::SinkExt;
+use std::sync::Arc;
 use utilities::Cast;
 
 /// Key-Value client.
 #[derive(Clone)]
 pub struct Kv {
     /// Etcd Key-Value client.
     client: KvClient,
+    /// Etcd client cache.
+    cache: Arc<Cache>,
+    /// Enable etcd client cache.
+    cache_enable: bool,
 }
 
+/// Etcd client cache default size.
+const ETCD_CACHE_DEFAULT_SIZE: usize = 64;
+
 impl Kv {
     /// Creates a new `KvClient`.
-    pub(crate) const fn new(client: KvClient) -> Self {
-        Self { client }
+    pub(crate) fn new(
+        client: KvClient,
+        watch_client: &WatchClient,
+        mut cache_size: usize,
```

Comment:
Does this "mut" mean user want to get cache_size?

Reply:
Removed mut here.